onepay/controllers/front/commit.php

```diff
@@ -15,16 +15,19 @@ public function postProcess() {
         if ($_SERVER['REQUEST_METHOD'] == 'GET') {
 
             $endpoint = Configuration::get('ONEPAY_ENDPOINT', null);
+            $apiKey = Configuration::get('ONEPAY_APIKEY', null);
+            $sharedSecret = Configuration::get('ONEPAY_SHARED_SECRET', null);
             
-            OnepayBase::setSharedSecret(Configuration::get('ONEPAY_SHARED_SECRET', null));
-            OnepayBase::setApiKey(Configuration::get('ONEPAY_APIKEY', null));
+            OnepayBase::setApiKey($apiKey);
+            OnepayBase::setSharedSecret($sharedSecret);
             OnepayBase::setCurrentIntegrationType($endpoint);
 
             $externalUniqueNumber = Tools::getValue('externalUniqueNumber');
             $occ = Tools::getValue('occ');
 
             try {
-                $options = new Options();
+
+                $options = new Options($apiKey, $sharedSecret);
```

Comment:
Why you need to set the apiKey and sharedSecret here too if you have just seted those before?

Reply:
is to omolog all the plugins, they are all like this

Reply:
Changed